### PR TITLE
Retry tests/test_write_cache in write back mode

### PR DESCRIPTION
**Pull request by bsbernd**
This test can fail whenever a something (kernel, userspace) decides to flush in between of two 2048B writes. These two writes are supposed to be merged into a single 4906 byte write by the kernel writeback cache, but _sometimes_ the test fails because 2048 byte writes get through.

### CODE REVIEW

File: test/test_write_cache.c [hunk 1]
```diff
@@ -51,6 +54,7 @@ static const struct fuse_opt option_spec[] = {
     FUSE_OPT_END
 };
 static int got_write;
+static atomic_int write_cnt;
```
Comment (ashleypittman) [inline test/test_write_cache.c:57]: Is this used anywhere?

**Review by ashleypittman — COMMENTED** [review]
Does the commit message need updating?

I've been testing this as well, the commit message talks about using two writes to check coalescing, using a greater number of writes has a higher chance of success.

File: test/test_write_cache.c (hunk 1 above)
Comment (bsbernd) [reply test/test_write_cache.c:57]: It has to - is a core requirement of this patch

```
assert(write_cnt < WRITE_SYSCALLS);
```

Comment (bsbernd) [reply test/test_write_cache.c:57]: Complete

```
    if (options.writeback)
        assert(write_cnt < WRITE_SYSCALLS);
    else
        assert(write_cnt == WRITE_SYSCALLS);
``

Comment (ashleypittman) [reply test/test_write_cache.c:57]: Ah yes, that makes much more sense, I'm not sure how I missed that.

**Review by ashleypittman — APPROVED** [review]
I was looking at the PR comment.

**Pull request merged**